src/libraries/Microsoft.Extensions.Options/src/OptionsCache.cs

```diff
@@ -31,9 +31,9 @@ public class OptionsCache<[DynamicallyAccessedMembers(Options.DynamicallyAccesse
         public virtual TOptions GetOrAdd(string? name, Func<TOptions> createOptions!!)
         {
             name ??= Options.DefaultName;
-            Lazy<TOptions>? value;
+            Lazy<TOptions> value;
 
-#if NETSTANDARD2_1
+#if NET || NETSTANDARD2_1
```

Comment:
Given that we could use TryGetValue in all frameworks, I assume that it is desirable to use GetOrAdd where we can. This way we'll use it for the newer .NET targets as well.